File: Regression/Section 4 - Simple Linear Regression/simpleLinearRegression.py
# Simple Linear Regression

# import libraries
import pandas as pa
import matplotlib.pyplot as plot
import numpy as nm

dataSet = pa.read_csv('Salary_Data.csv')
x = dataSet.iloc[:, :-1]
y = dataSet.iloc[:, 1]

# Split data into train and test set
from sklearn.model_selection import train_test_split
xTrain, xTest, yTrain, yTest = train_test_split(x, y, test_size = 1/3, random_state= 0)

# Fitting Simple Linear Regression
from sklearn.linear_model import LinearRegression
regressor = LinearRegression()
regressor.fit(xTrain, yTrain)
yPredict = regressor.predict(xTest)

# Visulaizing Predicted result
plot.scatter(xTrain, yTrain, color = 'RED')
plot.plot(xTrain, regressor.predict(xTrain), color = 'Blue')
plot.title('Experience VS Salary(Train Set)')
plot.xlabel('Experience')
plot.ylabel('Salary')
plot.show

plot.scatter(xTest, yTest, color = 'RED')
plot.plot(xTrain, regressor.predict(xTrain), color = 'Blue')
plot.title('Experience VS Salary(Test Set)')
plot.xlabel('Experience')
plot.ylabel('Salary')
plot.show

File: Regression/Section 4 - Simple Linear Regression/simpleLinearRegression.R
# Import dataset
data = read.csv('Salary_Data.csv')

# Split data into Train and test set
# install.packages('caTools')
library(caTools)
split = sample.split(data, SplitRatio = 2/3)
trainSet = subset(data, split == TRUE)
testSet = subset(data, split == FALSE)

# Fit the train set to linear regression model
regressor = lm(formula = Salary ~ YearsExperience,
               data = trainSet)

# Predict the Salary for test Set
salaryPredicted = predict(regressor, newdata = testSet)

# Visualize the Predicted values
# install.packages('ggplot2')
library(ggplot2)
ggplot()+
  geom_point(aes(x = trainSet$YearsExperience, y = trainSet$Salary), colour = 'RED')+
  geom_line(aes(x = trainSet$YearsExperience, y = predict(regressor,newdata = trainSet)), colour = 'BLUE')+
  ggtitle('Experience VS Salary (Train Set Result)')+
  xlab('Experience')+
  ylab('Salary')

# Train Set Result
library(ggplot2)
ggplot()+
  geom_point(aes(x = testSet$YearsExperience, y = testSet$Salary), colour = 'RED')+
  geom_line(aes(x = trainSet$YearsExperience, y = predict(regressor,newdata = trainSet)), colour = 'BLUE')+
  ggtitle('Experience VS Salary (Test Set Result)')+
  xlab('Experience')+
  ylab('Salary')